planner/core/rule_join_reorder_greedy.go

```diff
@@ -51,10 +55,62 @@ func (s *joinReorderGreedySolver) solve(joinNodePlans []LogicalPlan) (LogicalPla
 			cumCost: s.baseNodeCumCost(node),
 		})
 	}
+
+	for _, edge := range s.directedEdges {
+		nodeSet := s.directMap[edge[0]]
+		if nodeSet == nil {
+			nodeSet = make(map[LogicalPlan]struct{})
+			s.directMap[edge[0]] = nodeSet
+		}
+		nodeSet[edge[1]] = struct{}{}
+	}
+
+	isChanged := true
+	for isChanged {
+		isChanged = false
```

Comment:
```suggestion
	for isChanged := false; isChanged; isChanged {
```
And with the above comment, change `isChanged` to `isPropagated` may be better to understand.